packages/extism-tools/src/WikipediaTool.ts

```diff
@@ -0,0 +1,61 @@
+import createPlugin from "@extism/extism";
+import type { BaseTool, ToolMetadata } from "@llamaindex/core/llms";
+import type { JSONSchemaType } from "ajv";
+
+type WikipediaParameter = {
+  query: string;
+};
+
+export type WikipediaToolParams = {
+  metadata?: ToolMetadata<JSONSchemaType<WikipediaParameter>>;
+};
+
+const DEFAULT_META_DATA: ToolMetadata<JSONSchemaType<WikipediaParameter>> = {
+  name: "wikipedia_tool",
+  description: "A tool that uses a query engine to search Wikipedia.",
+  parameters: {
+    type: "object",
+    properties: {
+      query: {
+        type: "string",
+        description: "The query to search for",
+      },
+    },
+    required: ["query"],
+  },
+};
+
+export class WikipediaTool implements BaseTool<WikipediaParameter> {
```

Comment:
what do we need to do to make this more genric, like an `ExtismTool`?

Reply:
e.g. the how can we get the WASM function signature at run time and convert it to a JSON schema?